pylint/checkers/variables.py

```diff
@@ -2024,16 +2024,17 @@ def _store_type_annotation_names(self, node):
 
     def _check_self_cls_assign(self, node):
         """Check that self/cls don't get assigned"""
-        assign_names = {
-            target.name
-            for target in node.targets
-            if isinstance(target, nodes.AssignName)
-        }
+        assign_names = []
```

Comment:
```suggestion
        assign_names: Set[Union[str, None]] = set()
```
Why not use a `set` here?
Unfortunately `name` can still be `str | None`, although in practice it will always be `str`. Already deprecated in `astroid`.